scripts/version_discovery.py

```diff
@@ -0,0 +1,279 @@
+
+                if self.check_image_exists(image, tag):
+                    logger.info(f"Found matching image for {repo} at commit {sha[:8]}")
+                    return sha, tag
+
+            logger.warning(f"No commits found with matching images for {repo}/{image}")
+            return None
+        except Exception as e:
+            logger.error(f"Error finding commit with image for {repo}: {e}")
+            raise
+
+    def scan_components(self) -> list[RepositoryInfo]:
+        results: list[RepositoryInfo] = []
+        with ThreadPoolExecutor(max_workers=10) as executor:
+            future_to_repo = {
+                executor.submit(self.process_repository, repo, config): repo
+                for repo, config in REPOSITORIES.items()
+            }
+            for future in future_to_repo:
+                repo = future_to_repo[future]
+                try:
+                    repo_results = future.result()
+                    if repo_results:
+                        results.extend(repo_results)
+                except Exception as e:
+                    logger.error(f"Error processing {repo}: {e}")
+
+        return results
+
+    def process_repository(
+        self, repo: str, config: RepositoryConfig
+    ) -> list[RepositoryInfo]:
+        repo_results: list[RepositoryInfo] = []
+        version_prefix = config.get("version_prefix", "v")
+        images = config.get("images", [])
+
+        version_info = None
+        try:
+            version_info = self.find_latest_release_version(repo, version_prefix)
+        except Exception as e:
+            logger.error(f"Error finding latest release for {repo}: {e}")
+            
+        if version_info:
+            tag_name, _ = version_info
+            repo_results.append({
+                "repository": repo,
+                "ref": tag_name,
+                "image_url": None,
+            })
+        elif images:
+            for image in images:
+                component_name = f"{repo}/{image.split('/')[-1]}"
+                logger.info(f"Checking {component_name} latest commit")
+
+                commit_info = None
+                try:
+                    commit_info = self.find_latest_commit_with_image(repo, image)
+                except Exception as e:
+                    logger.error(f"Error finding commit with image for {repo}/{image}: {e}")
+                    
+                if commit_info:
+                    commit_sha, tag = commit_info
+                    repo_results.append({
+                        "repository": repo,
+                        "ref": commit_sha,
+                        "image_url": f"{image}:{tag}",
+                    })
+                else:
+                    logger.warning(f"No images found for {component_name}")
+        else:
+            logger.warning(f"No releases or tags found for {repo} and no images configured")
+
+        return repo_results
+
+    @staticmethod
+    def save_to_release_candidates(
```

Comment:
I think we need to extract release candidate loading and saving.
I would recommend repository pattern, where we would have ReleaseCandidateRepository and a ReleaseCandidate object (structured or unstructured)
For example, something like:
```
rc_repo = ReleaseCandidateRepository("release_candidates.yaml)
rc_list = rc_repo.list() 
rc_list.append(ReleaseCandidate({}))
rc_repo.save_list(rc_list)
```

The advantage is that we centralize the structure of this file: right now it's dispersed in multiple files and it makes it more difficult to understa.d